src/contracts/test/EventEmitter.sol

```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: LGPL-3.0-or-newer
+pragma solidity ^0.7.5;
+
+contract EventEmitter {
```

Comment:
Its a pitty the `waffle` MockContract can't do this. In theory, [we should be able to](https://ethereum-waffle.readthedocs.io/en/latest/matchers.html#called-on-contract-with-arguments)

```ts
expect('mockContractFunction').to.be.calledOnContractWith(mockContract, [parameters]);
```

But this doesn't work with `waffle` + `hardhat` :cry:

Reply:
Yeah, it would be nice to have it mockable. Another issue is that I believe `calledOnContractWith` cannot account for the order in which transactions were executed, and in our case this is very important.